File: 1899-count-items-matching-a-rule/count-items-matching-a-rule.py
class Solution:
    def countMatches(self, items: List[List[str]], ruleKey: str, ruleValue: str) -> int:
        count = 0
        for i in items:
            if ruleKey == "type":
                count += 1 if i[0] == ruleValue else 0;
            elif ruleKey == "color":
                count += 1 if i[1] == ruleValue else 0;
            elif ruleKey == "name":
                count += 1 if i[2] == ruleValue else 0;
        return count;
        

File: 1899-count-items-matching-a-rule/count-items-matching-a-rule.go
func ternary(a, b string) int{
    if a == b{
        return 1
    }
    return 0;
}

func countMatches(items [][]string, ruleKey string, ruleValue string) int {
    count := 0;
    for _, item := range items{
        if ruleKey == "type"{
            count += ternary(item[0], ruleValue);
        }else if ruleKey == "color"{
            count += ternary(item[1], ruleValue);
        }else{
            count += ternary(item[2], ruleValue);
        }
    }
    return count;
}